website/website/add_youtuber.py

```diff
@@ -0,0 +1,77 @@
+from googleapiclient.discovery import build
+from urllib.parse import urlparse, unquote
+import os
+API_KEY = os.environ.get('API_KEY')
+
+
+class YoutubeApi:
+    """A class to interact with the YouTube API."""
+
+    def __init__(self, API_KEY, channel_url) -> None:
+        self.API_KEY = API_KEY
+        self.channel_url = channel_url
+        self.username = None
+        self.channel_id = None
+        self.channel_title = None
+        self.channel_description = None
+
+    def get_username_from_url(self, channel_url: str) -> str:
+        """
+        Extract the username from the channel URL and return it.
+
+        Parameters:
+        channel_url (str): The URL of the YouTube channel.
+
+        Returns:
+        str: The username extracted from the channel URL.
+
+        """
+        parsed_url = urlparse(channel_url)
+        if parsed_url.path.startswith("/channel/"):
```

Comment:
To me personally, the related if-elif-else blocks are not readable.

I'd prefer to see every if as independant statement, if possible:

```
if parsed_url.path.startswith("/channel/"):
    self.username = unquote(parsed_url.path[9:])
    return

if parsed_url.path.startswith("/c/"):
    self.username = unquote(parsed_url.path[3:])
    return

...

```

I'm not not insisting.

Reply:
In this case, we always will do all checks, which will be slower.
But I changed this part of the code and moved the case with '@' on the first position, which is most often seen.